src/test/java/com/google/devtools/build/lib/analysis/config/transitions/ComposingTransitionFactoryTest.java

```diff
@@ -0,0 +1,233 @@
+package com.google.devtools.build.lib.analysis.config.transitions;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.devtools.build.lib.analysis.config.BuildOptions;
+import com.google.devtools.build.lib.analysis.config.HostTransition;
+import com.google.devtools.build.lib.analysis.config.TransitionFactories;
+import com.google.devtools.build.lib.cmdline.Label;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link ComposingTransitionFactory}. */
+@RunWith(JUnit4.class)
+public class ComposingTransitionFactoryTest {
+  // Use starlark flags for the test since they are easy to set and check.
+  private static final Label FLAG_1 = Label.parseAbsoluteUnchecked("//flag1");
+  private static final Label FLAG_2 = Label.parseAbsoluteUnchecked("//flag2");
+
+  @Test
+  public void compose_patch_patch() {
+    // Same flag, will overwrite.
+    TransitionFactory<StubData> composed =
+        ComposingTransitionFactory.of(
+            TransitionFactories.of(new StubPatch(FLAG_1, "value1")),
```

Comment:
What's a StubPatch?

Reply:
It's a small class defined further down in the test file in order to provide a simple patch transition for testing.

Reply:
Indeed, I missed it!